dpctl/tensor/_reduction.py

```diff
@@ -123,6 +123,10 @@ def sum(arr, axis=None, dtype=None, keepdims=False):
 
     res_usm_type = arr.usm_type
     if arr.size == 0:
+        if keepdims:
+            res_shape = res_shape + (1,) * red_nd
+            inv_perm = sorted(range(nd), key=lambda d: perm[d])
```

Comment:
shouldn't we move that above the `if` to reduce code duplication, since it looks common and without any dependency on logic below?

Reply:
Perhaps in a separate PR. We are talking about deduplicating two lines of code at the expense of making the code more complex, since unexpanded `res_shape` is needed in the call to `sum_over_axis`. 

Thank you for the review!